PRD-994 Amend error message for abort reject

HarishRocks opened the pull request.

HarishRocks committed:
PRD-994 Amend error message for abort reject

irshadCypherock commented:
Task linked: [CSI-1150 Abort Process and handle if device is busy](https://app.clickup.com/t/37308523/CSI-1150)

HarishRocks committed:
CSI-1150 Check if abortable before aborting

HarishRocks committed:
CSI-1150 check if process aborted

If a flow is about to request for a lesser sequence number than the current one. It is concluded that this process was aborted and must stop requesting.

irshadCypherock commented on src/core/deviceConnection/index.ts:
```diff
@@ -449,14 +454,23 @@ export class DeviceConnection
-    });
-
-    return resp;
+    const status = await this.getStatus({ maxTries: 2 });
+    if (
+      status.deviceIdleState === DeviceIdleState.USB &&
+      !status.abortDisabled
+    ) {
+      const resp = await operations.sendAbort({
+        connection: this,
+        version,
+        sequenceNumber: params.sequenceNumber,
+        maxTries: params.maxTries
+      });
+
+      return resp;
+    } else {
+      logger.info("Abort rejected based on device's current status ");
+      logger.info(status);
+    }
```
Can you explore the possibility of storing the `abortDisabled` variable in the device connection itself, and use it when required?

So whenever we get a new status from device, we update the `abortDisabled` variable and use it in sendAbort function.

irshadCypherock commented on src/core/operations/waitForPacket.ts:
```diff
@@ -70,6 +70,14 @@ export const waitForPacket = ({
               );
 
               const rejectStatus = parseInt(`0x${rawData}`, 16);
+              if (
+                rejectStatus === ErrorPacketRejectReason.INVALID_SEQUENCE_NO
+              ) {
+                error = new DeviceError(
+                  DeviceErrorType.PROCESS_ABORTED_BY_USER
+                );
+                break;
+              }
```
Add a check for increased sequence number to determine if the flow was aborted by cysync or it's an actual error

irshadCypherock reviewed (changes requested):
Need to add sequence number increase check to `sendCommand` also.

irshadCypherock committed:
Merge branch 'beta' into CSI-1150_Abort-Process-and-handle-if-device-is-busy

irshadCypherock committed:
CSI-1150: Handling cancel flow
